### ISSUE TITLE
Markdown for tasks does not support indent

### ISSUE THREAD

**Issue by mandyWW** [buggyrace/buggy-race-server#204]
Markdown with indents is not reflected in the task list when uploaded. For example, the following text will not be formatted nicely.

* It's best if you run in Flask's _development environment_ so that any changes you make to your code will be immediately reflected in the browser. To do that, you need to set the 
environment variable `FLASK_ENV` to `development`before you run `app.py`.
Once you've done this, it's good for the rest of the session.

   In the VSCode terminal type `export FLASK_ENV=development`.  

   When you get to task [3-ENV](http://rhul.buggyrace.net/project/tasks/3-ENV)
you'll investigate other ways of doing this.

**Comment by davewhiteland** [comment]
I don't think this is a bug — it's a limitation of markdown (btw, the markdown conversion the race server is using is the [PyPi markdown](https://pypi.org/project/Markdown/), which is very widely-used). 

I don't think there is a clean way in markdown to break a bullet list over several lines, _including empty lines_ (because although the double-space-at-end-of-line markdown causes a line-break, if there's _only_ whitespace on the line, markdown interprets that as the empty line which is ending the bullet list).

But... you _can_ trick markdown into thinking a line is not empty (so doesn't end the indented block of a list item) by putting a space in backticks on the indented line, and then immediately ending the line with two spaces.

However, in the case of the task list, because this markdown is being rendered as text which itself is being presented to the browser, you can also drop HTML in there: either use `<br>` for a linebreak, or `&nbsp;` for the "non-breaking space": (note that to demonstrate the `&nbsp;` there are (invisible!) two spaces on the end of that line and one above it.

Here are all these dirty techniques being demonstrated:

```markdown
* See the [tech note about localhost](%BUGGY_RACE_SERVER_URL%/tech_notes/localhost).

* It's best if you run in Flask's _development environment_ so that any changes you make to your code will be immediately reflected in the browser. To do that, you need to set the 
environment variable `FLASK_ENV` to `development` before you run `app.py`.
Once you've done this, it's good for the rest of the session.
  <br><br>
  In the VSCode terminal type `export FLASK_ENV=development`.  
  &nbsp;  
   When you get to task [3-ENV](http://rhul.buggyrace.net/project/tasks/3-ENV)
you'll investigate other ways of doing this.

* Next bullet point...  
  ` `  
  ...this is after one "empty": line, but still indented as
  part of the bullet point text. That works because the backticks
  are delimiting a space which markdown won't collapse, which itself
  is then followed by two empty spaces (marking a linebreak).
  Tricksy!
```

Here's a screenshot of that last bit with invisibles shown: note the two empty spaces on the end-of-line, which markdown renders as a linebreak if it can:

![Screenshot 2024-03-26 at 13 57 48](https://github.com/buggyrace/buggy-race-server/assets/111320/8c9637d9-e146-4589-ba12-bda9d340ccaf)


...whole example resulting in:

![Screenshot 2024-03-26 at 13 53 18](https://github.com/buggyrace/buggy-race-server/assets/111320/be8f2310-e0ac-41c5-ba88-d62edc3ecc1f)
